django/contrib/auth/management/commands/changepassword.py

```diff
@@ -35,13 +43,30 @@ def add_arguments(self, parser):
             choices=tuple(connections),
             help='Specifies the database to use. Default is "default".',
         )
+        parser.add_argument(
+            "--stdin",
+            action="store_true",
+            help="Read new password from stdin rather than prompting, default is False",
+        )
 
     def handle(self, *args, **options):
+        def _input_getter_getpass():
+            p1 = self._get_pass()
+            p2 = self._get_pass("Password (again): ")
+            return p1, p2
+
+        def _input_getter_stdin():
+            return self._get_stdin()
+
         if options["username"]:
             username = options["username"]
         else:
             username = getpass.getuser()
 
+        if options["stdin"]:
+            input_getter = _input_getter_stdin
+        else:
+            input_getter = _input_getter_getpass
```

Comment:
Seems like there might be an opportunity here to implement `--no-input` to rhyme with the other commands. Either `--stdin` implies `--no-input`, or we just merge the two options. I'm not sure. WDYT?